.github/workflows/release-notes.yml

```diff
@@ -20,4 +19,4 @@ jobs:
     - name: Check release notes
       run: |
         git fetch origin main
-        git diff origin/main --exit-code include/ceed/* && ! git diff origin/main --exit-code doc/sphinx/source/releasenotes.rst
+        ( ! git diff origin/main --exit-code include/ceed/* ) || git diff origin/main --exit-code doc/sphinx/source/releasenotes.rst
```

Comment:
Maybe this would be easier to interpret and maintain?
```suggestion
        if git diff origin/main --exit-code include/ceed/*; then
            echo "No public interface changes detected"
        elif ! git diff origin/main --exit-code doc/sphinx/source/releasenotes.rst
            echo "API changes detected, but release notes not updated"
            exit 1
        fi
```

Reply:
Agreed, this is better.